native/src/main/java/io/ballerina/observe/trace/newrelic/NewRelicTracerProvider.java

```diff
@@ -0,0 +1,116 @@
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+
+import java.io.PrintStream;
+import java.util.concurrent.TimeUnit;
+
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME;
+
+/**
+ * This is the New Relic tracing extension class for {@link TracerProvider}.
+ */
+public class NewRelicTracerProvider implements TracerProvider {
+    private static final String TRACER_NAME = "newrelic";
+    private static final String TRACE_REPORTER_ENDPOINT = "https://otlp.nr-data.net:4317";
+    private static final String TRACE_API_KEY_HEADER = "Api-Key";
+
+    static SdkTracerProviderBuilder tracerProviderBuilder;
+
+    @Override
+    public String getName() {
+        return TRACER_NAME;
+    }
+
+    @Override
+    public void init() {
+        // Do Nothing
+    }
+
+    public static BArray startPublishingTraces(BString apiKey, BString samplerType, BDecimal samplerParam,
+                                                int reporterFlushInterval, int reporterBufferSize) {
+        BArray output = ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_STRING));
+
+        OtlpGrpcSpanExporter exporter = OtlpGrpcSpanExporter.builder()
+                .setEndpoint(TRACE_REPORTER_ENDPOINT)
+                .addHeader(TRACE_API_KEY_HEADER, apiKey.getValue())
+                .build();
+
+        tracerProviderBuilder = SdkTracerProvider.builder()
+                .addSpanProcessor(BatchSpanProcessor
+                        .builder(exporter)
+                        .setMaxExportBatchSize(reporterBufferSize)
+                        .setExporterTimeout(reporterFlushInterval, TimeUnit.MILLISECONDS)
+                        .build());
+
+        tracerProviderBuilder.setSampler(selectSampler(samplerType, samplerParam));
+        output.append(StringUtils.fromString("ballerina: started publishing traces to New Relic on " +
+                TRACE_REPORTER_ENDPOINT));
+
+        return output;
+    }
+
+    private static Sampler selectSampler(BString samplerType, BDecimal samplerParam) {
+        switch (samplerType.getValue()) {
+            default:
+            case "const":
+                if (samplerParam.value().intValue() == 0) {
+                    return Sampler.alwaysOff();
+                } else {
+                    return Sampler.alwaysOn();
+                }
+            case "probabilistic":
+                return Sampler.traceIdRatioBased(samplerParam.value().doubleValue());
+            case RateLimitingSampler.TYPE:
+                return new RateLimitingSampler(samplerParam.value().intValue());
+        }
+    }
+
+    @Override
+    public Tracer getTracer(String serviceName) {
+
+        return tracerProviderBuilder.setResource(
+                        Resource.create(Attributes.of(SERVICE_NAME, serviceName)))
+                .build().get("newrelic");
```

Comment:
This is same as [here](https://github.com/ballerina-platform/module-ballerinax-newrelic/pull/1/files#diff-dfb02ee96cce2ac78b9c9c5898f7a387b45d786a737fdd3535209be2373b7a6aR49), can we reuse by moving into a constant file?

Reply:
There is a constant here 
https://github.com/ballerina-platform/module-ballerinax-newrelic/pull/1/files#diff-6f3d8b6d22f0aaa36723e3f6339564500c33696e111e7d1503879fd2722eb7fbR32

Reply:
Addressed review suggestions